### PR TITLE
feat: show api errors on kre version website

### LINKED ISSUE #297: Add error status to version
Sometimes it is not possible to create a version and we do not have an **ERROR state and error message** for those cases.

Current version statuses:
```
enum VersionStatus {
  CREATING
  CREATED
  STARTING
  STARTED
  PUBLISHED
  STOPPING
  STOPPED
}
```

Change WatchVersionStatus subscription to WatchVersion, adding to Version entity a error field to save the error message.
image

### PR DESCRIPTION
**Pull request by Sergiodfdez**
Close #297

### CODE REVIEW

File: admin/admin-api/adapter/repository/mongodb/version.go [hunk 1]
```diff
@@ -135,3 +135,20 @@ func (r *VersionRepoMongoDB) SetStatus(ctx context.Context, versionID string, st
 
 	return nil
 }
+
+func (r *VersionRepoMongoDB) SetError(ctx context.Context, version *entity.Version, errorMessages []string) (*entity.Version, error) {
```
Comment (guumaster) [inline admin/admin-api/adapter/repository/mongodb/version.go:139]: Maybe is more clear `SetErrors` to reflect that is receiving an array and not a single error.

File: admin/admin-ui/src/Pages/Runtime/pages/RuntimeVersions/components/VersionInfo/VersionInfo.tsx [hunk 2]
```diff
@@ -19,13 +19,17 @@ function VersionInfo({ version }: Props) {
   const { runtimeId } = useParams();
 
   const isPublishedVersion = version.status === VersionStatus.PUBLISHED;
+  const isErrorState = version.status === VersionStatus.ERROR;
```
Comment (mgs95) [inline admin/admin-ui/src/Pages/Runtime/pages/RuntimeVersions/components/VersionInfo/VersionInfo.tsx:22]: should be renamed to `hasError` for a more consistent meaning

File: admin/admin-ui/src/Pages/Runtime/pages/RuntimeVersions/components/VersionInfo/VersionInfo.tsx [hunk 3]
```diff
@@ -19,13 +19,17 @@ function VersionInfo({ version }: Props) {
...
 
   const versionPath = buildRoute.version(
     ROUTE.RUNTIME_VERSION_STATUS,
     runtimeId,
     version.id
   );
 
+  const errors = version.errors.map((error: string) => (
+    <p className={styles.versionError}>{error}</p>
```
Comment (mgs95) [inline admin/admin-ui/src/Pages/Runtime/pages/RuntimeVersions/components/VersionInfo/VersionInfo.tsx:31]: You forgot to add a `key` to this element here.

File: admin/admin-ui/src/Pages/Runtime/pages/RuntimeVersions/components/VersionInfo/VersionInfo.module.scss [hunk 4]
```diff
@@ -155,6 +155,11 @@ $color-border: palette(base, 400);
 
   .description {
     @include font-small;
+    margin-bottom: 8px;
+  }
+
+  .versionError {
+    color: red;
```
Comment (mgs95) [inline admin/admin-ui/src/Pages/Runtime/pages/RuntimeVersions/components/VersionInfo/VersionInfo.module.scss:162]: palette('error')

**Review by mgs95 — APPROVED** [review]

**Review by guumaster — APPROVED** [review]

**Pull request merged**